### DISCUSSION TITLE
Incorrect filling of areal objects

### DISCUSSION THREAD

**Discussion by jonimnim** [mapsforge/mapsforge#1271] [Q&A] [answered]
Hello. I have a problem with filling area objects using mapsforge. Tell me what am I doing wrong?
I have an object file in the OSM format ("v6.osm"). [send.zip](https://github.com/mapsforge/mapsforge/files/7667809/send.zip)
I convert it so that it can be understood by osmosis.
`osmconvert v6.osm --fake-author -o=v66.osm`
Then, using osmosis, I convert it to a file for mapsforge.
`osmosis --rx file="v66.osm" --sort --mw file="v6.map" bbox=59.9313911574306886,30.3582604281478794,59.9342191007421121,30.3627518418694216 tag-conf-file="tags_a.xml"`
tag-conf-file="tags_a.xml" - file with my tags.
I am trying to view this file with the theme "testTheme.xml".
but I get what is in the picture
![изображение](https://user-images.githubusercontent.com/16680576/145019809-262f109e-3678-4666-bf22-f127e4b3e5fe.png)
And when scaling, the colors change randomly.
Please help me solve this problem.

**Comment by devemux86** [comment, marked as answer]
Use a separate / specific tag to categorize the elements (not the `name`).
The [TagTransform](https://wiki.openstreetmap.org/wiki/Osmosis/TagTransform) Osmosis plugin may help with custom tags.

See Mapsforge [tag-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml) and [themes](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-themes/src/main/resources/assets/mapsforge) for more details.

![Cruiser](https://user-images.githubusercontent.com/3484020/145025498-f05d9ff7-72c8-4d5f-adda-6e3442d9fec0.png)

**Comment by jonimnim** [comment]
Thanks a lot. They helped out. Otherwise I have been suffering with this for two days and cannot understand what is happening.

**Comment by jonimnim** [comment]
Hello. Again there was a problem with incorrect filling, but of a slightly different plan. Now it is pouring what should not be poured. Those. there is a polygon with a hole, but for some reason the hole at the polygon is also flooded. Please help me figure it out.
In the file I attach osm files and a map file that turned out, as well as a theme and a file with tags
[mapsforge_island.zip](https://github.com/mapsforge/mapsforge/files/7987392/mapsforge_island.zip)
.

**Reply by devemux86** [reply]
_Please create new discussions for new questions._

---

If you mean maps with coastlines, see the [map creation guide](https://github.com/mapsforge/mapsforge/blob/master/docs/MapCreation.md) about how to include sea + land + map data.
